go.mod

```diff
@@ -69,3 +69,5 @@ require (
 	gopkg.in/bblfsh/sdk.v1 v1.17.0
 	gopkg.in/src-d/go-errors.v1 v1.0.0
 )
+
+replace github.com/bblfsh/sdk/v3 => github.com/dennwc/sdk/v3 v3.0.0-20190515204145-5c71eed53fbf
```

Comment:
Nice! Need not to forget to remove it later 😆 

Just curious - was this one added with a text editor, or is it some tooling that was used?

Reply:
Sure :) It's nice that it's possible to test CI this way, so I can work on changes to all our project at the same time and then test every part in CI before even finishing the review process. This saves a lot of time.

Can be added with a text editor as:
```
replace github.com/bblfsh/sdk/v3 => github.com/dennwc/sdk/v3 some_branch
```
and then `go mod tidy` will rewrite it to a specific revision. The command may be different since in my case it's done automatically by IDE.

And the second way is to use `go mod` command to add this line:
```
go mod edit -replace github.com/bblfsh/sdk/v3=github.com/dennwc/sdk/v3@some_branch
```

Reply:
It's worth noting that you can even use `replace` to point to a local file path, if you want. I do this sometimes when I am preparing dependent changes in my personal repositories where I don't want to push the fixes upstream yet.

```
replace github.com/foo/bar => /Users/michael/local/foo/bar
```

Reply:
In re: "Not forgetting to remove patches later":

We could probably add a presubmit check for this, either as a hook or in CI. I'd say it's fair that the committed `go.mod` should not contain any `replace` lines except for work-in-flight. It would be pretty easy to write a tool that flags an error for any such line that isn't accompanied by a comment pointing to an open PR (or some such). Then the CI would protect us. If you think that's worthwhile, I can write it up.

Reply:
Sorry, I think I have not communicated my intention well here.

The whole point of me pushing this PR with the `replace` was to test if CI fails with all the changes I've made without waiting for SDK PR to be merged. I'm aware that the line must be removed, and putting the CI check against those `replace` directives would defeat the purpose of doing the PR in the first place. Locally I test with a relative `replace` (`../sdk`), but I assume my environment is not "clean" and there may be issues with the change that CI will catch.

Reply:
> Locally I test with a relative `replace` (`../sdk`), but I assume my environment is not "clean" and there may be issues with the change that CI will catch.

Right—that makes sense. I think I'm the one who confused things: What I was trying to describe is a check that prevents a PR from being _merged_ with the replacement still in place. Putting it in the context of CI wasn't really relevant.

Reply:
Hmm, preventing the merge actually sound interesting, but we'll probably need a separate Github Check for this. Maybe it can be implemented as Lookout analyzer :)